Modules/posixmodule.c

```diff
@@ -8263,42 +8241,25 @@ os_setpgrp_impl(PyObject *module)
 #ifdef HAVE_GETPPID
 
 #ifdef MS_WINDOWS
-#include <tlhelp32.h>
+#include <processsnapshot.h>
 
 static PyObject*
 win32_getppid()
 {
-    HANDLE snapshot;
     PyObject* result = NULL;
-    BOOL have_record;
-    PROCESSENTRY32 pe;
-
-    DWORD mypid = GetCurrentProcessId(); /* This function never fails */
+    HANDLE myhandle = GetCurrentProcess();
 
-    snapshot = CreateToolhelp32Snapshot(TH32CS_SNAPPROCESS, 0);
-    if (snapshot == INVALID_HANDLE_VALUE)
+    HPSS snapshot = NULL;
+    if (PssCaptureSnapshot(myhandle, PSS_CAPTURE_NONE, 0, &snapshot) != ERROR_SUCCESS)
         return PyErr_SetFromWindowsErr(GetLastError());
```

Comment:
Use [PEP 7](https://peps.python.org/pep-0007/) style. Also, you need to store the resulting error code with the snapshot API instead of calling `GetLastError()`. For example:

```c
    DWORD error;
    HANDLE process = GetCurrentProcess();
    
    error = PssCaptureSnapshot(process, PSS_CAPTURE_NONE, 0, &snapshot);
    if (error != ERROR_SUCCESS) {
        return PyErr_SetFromWindowsErr(error);
    }
```